### DISCUSSION TITLE
Multiple swichting domains

### DISCUSSION THREAD

**Discussion by rainerduffner** [netbox-community/netbox#17190] [General]
Hi,

we have multiple switching domains currently, due to legacy design decisions and acquisitions over the year.

I know one is supposed to create "fake VRFs" for that - but our networking department does not want to go down that route because we also have lots of "actual" VRFs and the fear is that it will create a mess right from the beginning.

We are an MSP and have a home-grown IPAM inside our home-grown billing and provisioning system. The IPAM hasn't been fit for purpose for a long time, but adapting it to IPv6 is impossible, neither does it have an API and thus the decision to venture to something new.

We have about 20 years of legacy in that system that my co-worker and myself are slowly untangling (we are both crazy enough and I am long-tenured enough to believe we can do this). We are unix sysadmins, not networking.

The switching domains aren't always aligned to customers or companies but more along purpose (there's backup, storage, "most of the rest", on-premise clouds).

Is there really no way to represent this?

We were planing to re-use tenant groups and tenants to represent our reseller and customer model from the old tool, as that information has to be kept in sync between billing and Netbox.

**Comment by candlerb** [comment]
What do you mean by "switching domain"? Do you mean a region of your network where vlans 100, 200, 300 are different from vlan 100, 200, 300 in a different region?  VRFs are a layer 3 concept.

If you mean regions of your network where IP address allocations overlap between them but not within them, then it has been long ago suggested to separate "network namespaces" or "L3 domain" from "VRF", but nothing has happened. e.g. #7608 but I can't find anything more concrete right now.
